docs/api.yaml

```diff
@@ -0,0 +1,925 @@
+        default:
+          $ref: '#/components/responses/error_response'
+          
+    put:
+      description: >-
+        Unsideline messages from DLT for further consumption.
+      tags: 
+        - Consume
+      responses:
+        '200':
+          description: OK
+        default:
+          $ref: '#/components/responses/error_response'
+   
+     
+components:
+  schemas: 
+    tenant:
+      type: object
+      properties:
+        name:
+          type: string
+          example: "finance-project"
+        quota:
+          type: object
+          # TODO:: To be enabled later, based on enforcement static/dynamic. 
+          # properties:
+          #   storage:
+          #     type: integer
+          #   throughput:
+          #     type: integer
+        
+    tenants:
+      type: array
+      items:
+        $ref: '#/components/schemas/tenant'
+
+    property_bag:
+      type: object
+      additionalProperties:
+        type: string
+        
+    retention_policy:
+      type: object
+      properties:
+        retention_hours:
+          type: integer
+          example: 24
+          default: 48
+          maximum: 76
+          
+    capacity_policy:
+      type: object
+      properties:
+        max_throughput:
+          type: integer
+          description: "Maximum throughput in Mbps."
+          example: 10
+        max_storage:
+          type: integer
+          description: "Maximum size in MB"
+          example: 1500
+        message_size:
+          type: integer
+          description: "Maximum message size in KBs."
+          example: 10
+          
+    retry_policy:
+      type: object
+      properties:
+        retry_codes:
+          type: array
+          items:
+            type: integer
+          example: [500, 501]
+        kind:
+          type: string
+          example: "exponential"
+          enum: 
+            - "exponential"
+            - "linear"
+        minbackoff:
+          type: integer
+          description: "minimum retry interval in ms."
+          example: 300
+        maxbackoff:
+          type: integer
+          description: "maxmimum retry interval in ms."
+          example: 1500
+        multiplier:
+          type: integer
+          description: "step for iteration"
+          example: 10
+        retry_attempts:
+          type: integer
+          description: "maximum retry attempts allowed."
+          maximum: 3
+          
+    dlq_policy:
+      type: object
+      properties:
+        enabled:
+          type: boolean
+          example: true
+        endpoint:
+          $ref: '#/components/schemas/endpoint'
+          
+    consumption_policy:
+      type: object
+      properties:
+        max_parallelism:
+          type: integer
+          example: 10
+        recovery_allocation:
+          type: number
+          example: 0.5
+        dlq_from_recovery:
+          type: number
+          example: 0.5
+        max_error_threshold:
```

Comment:
In case of grouped subscription, will the error_rate be calculated in terms of failed groups, instead of messages failing to RQ / DLQ. right?

Reply:
I was thinking of later.. as primarily MTTR and also rq/dq size will be determined by number of messages.